hathor/p2p/netfilter/rule.py

```diff
@@ -28,6 +28,32 @@ def __init__(self, match: 'NetfilterMatch', target: 'NetfilterTarget'):
         self.match = match
         self.target = target
 
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, NetfilterRule):
+            return NotImplemented
+
+        self_chain_name = self.chain.name if self.chain else None
+        other_chain_name = other.chain.name if other.chain else None
+        return self_chain_name == other_chain_name and self.match == other.match and self.target == other.target
+
+    def to_json(self) -> Dict[str, Any]:
+        data: Dict[str, Any] = {}
+
+        if self.chain:
```

Comment:
Why don't we let the chain object serializes itself into a json?

Reply:
Added a to_json for each class